csv-validator-ui/src/main/scala/uk/gov/nationalarchives/csv/validator/ui/ScalaSwingHelpers.scala

```diff
@@ -42,17 +53,21 @@ object ScalaSwingHelpers {
    * @param result A function which takes the chosen file
    * @param locateOver A component over which the FileChooser dialog should be located
    */
-  def chooseFile(fileChooser: FileChooser, result: Path => Option[IOException], locateOver: Component, dialogText: String = "Save") : Unit = {
-    fileChooser.showDialog(locateOver, dialogText) match {
-      case Result.Approve =>
-        result(fileChooser.selectedFile.toPath) match {
-          case Some(ioe) =>
-            ioe.printStackTrace()
-            Dialog.showMessage(fileChooser, s"${ioe.getClass.getName}: ${ioe.getMessage}", "Unable to Save file", Message.Error)
-          case None =>
-        }
-      case Result.Cancel =>
-    }
+  def chooseFile(fileChooser: FileChooser, result: Path => Try[String], locateOver: Component, dialogText: String = "OK") : Unit = {
+    val showDialogAction = fileChooser.showDialog(locateOver, dialogText)
```

Comment:
dialogText usually refers to the title of the dialog, in the past that was being reused for the button. Now that you have properly separated it out, you could rename this `dialogText` variable to something like `approveBtnText` or `approveText` or `okBtnText` something like that

Reply:
I'll do `approveBtnText`, so that I can do the same for `saveFile` and since "OK" is only a default param